core/src/com/unciv/models/ruleset/unique/Unique.kt

```diff
@@ -301,8 +301,23 @@ class LocalUniqueCache(val cache:Boolean = true) {
     // This stores sequences *that iterate directly on a list* - that is, pre-resolved
     private val keyToUniques = HashMap<String, Sequence<Unique>>()
 
+    fun forCityGetMatchingUniques(city: City, uniqueType: UniqueType) : Sequence<Unique> {
+        return forCityGetMatchingUniques(city, uniqueType, StateForConditionals(city.civ, city))
+    }
+    fun forCityGetMatchingUniques(city: City, uniqueType: UniqueType, stateForConditionals: StateForConditionals) : Sequence<Unique> {
+        return get("city-${city.id}-${uniqueType.name}-${stateForConditionals}", city.getMatchingUniques(uniqueType, stateForConditionals))
+    }
+
+    fun forCivGetMatchingUniques(civ: Civilization, uniqueType: UniqueType) : Sequence<Unique> {
```

Comment:
state for conditionals should be an optional parameter defaulting to this and then we don't need 2 functions
Same for city functions